app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt

```diff
@@ -85,18 +81,24 @@ class TrayPagerAdapter(
             SyncedTabsPageViewHolder.LAYOUT_ID -> {
                 SyncedTabsPageViewHolder(
                     itemView,
-                    tabsTrayStore
+                    tabsTrayStore,
+                    TabClickDelegate(navInteractor)
                 )
             }
             else -> throw IllegalStateException("Unknown viewType.")
         }
     }
 
+    /**
+     * Until [TrayPagerAdapter] is replaced with a Compose implementation, some sort of adapter
+     * needs to be passed to [SyncedTabsPageViewHolder] for the abstract class contract.
+     * For more details: https://github.com/mozilla-mobile/fenix/issues/21318
+     */
     override fun onBindViewHolder(viewHolder: AbstractPageViewHolder, position: Int) {
         val adapter = when (position) {
             POSITION_NORMAL_TABS -> normalAdapter
             POSITION_PRIVATE_TABS -> privateAdapter
-            POSITION_SYNCED_TABS -> syncedTabsAdapter
+            POSITION_SYNCED_TABS -> privateAdapter
```

Comment:
We could just take out this line and the throw below, instead of adding side-effects.